Server/ReasnAPI/ReasnAPI/Services/ImageService.cs

```diff
@@ -1,16 +1,156 @@
 ﻿using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+using ReasnAPI.Models.Enums;
+using ReasnAPI.Models.Mappers;
+using static System.Net.Mime.MediaTypeNames;
+using Image = ReasnAPI.Models.Database.Image;
+using ReasnAPI.Exceptions;
+
+namespace ReasnAPI.Services;
+public class ImageService(ReasnContext context)
+{
+    public List<ImageDto> CreateImages(List<ImageDto> imageDtos)
+    {
+        var newImages = new List<Image>();
+
+        foreach (var imageDto in imageDtos)
+        {
+            var image = context.Images.FirstOrDefault(r => r.ObjectId == imageDto.ObjectId && r.ObjectType == imageDto.ObjectType && r.ImageData == imageDto.ImageData);
+            if (image is not null)
+            {
+                continue;
+            }
+
+            var newImage = imageDto.ToEntity();
+            
+            newImages.Add(newImage);
+        }
+
+        if (newImages.Any())
+        {
+            var objectType = newImages[0].ObjectType;
+          
+        
+            if (objectType == ObjectType.User && newImages.Count == 1)
+            {
+                context.Images.Add(newImages[0]);
+            }

```

Comment:
Here you do nothing when count is greater than 0 and object type is user

Reply:
So shouldn't we throw an error there too?

Reply:
yes sir